### ISSUE TITLE
Anti-Sandbox Check - Known file names

### ISSUE THREAD

**Issue by recvfrom** [LordNoteworthy/al-khaser#188]
Gootkit will self-destruct if it's running under one of the following file names (presumably as an anti-sandbox check):
```
SAMPLE.EXE
MALWARE.EXE
BOT.EXE
SANDBOX.EXE
TEST.EXE
KLAVME.EXE
MYAPP.EXE
TESTAPP.EXE
```
Reference: https://www.sentinelone.com/blog/gootkit-banking-trojan-deep-dive-anti-analysis-features/

Emotet looks for:
```
sample.*
mlwr_smple.*
artifact.exe
```
Reference: https://blog.trendmicro.com/trendlabs-security-intelligence/new-emotet-hijacks-windows-api-evades-sandbox-analysis/

It'd be cool if al-khaser had this check as well, for cases where someone uses the al-khaser binary to test out a sandbox.

**Comment by LordNoteworthy** [comment]
Hey @recvfrom 

Sounds good. Can you submit a PR or shall I do it?

Basically we need to call GetModuleFleName() then grab the filename from the full path using PathFindFileName() or PathStripPath().

**Comment by mq1n** [comment]
> Hey @recvfrom
> 
> Sounds good. Can you submit a PR or shall I do it?
> 
> Basically we need to call GetModuleFleName() then grab the filename from the full path using PathFindFileName() or PathStripPath().

maybe walking in PEB can be more useful,

**Comment by LordNoteworthy** [comment]
@mq1n using the PEB is also a way, though I am wondering if that's gonna make it any stealthier? Because my guess is that automated scanning systems just renamed the sample on disk and not by hooking GetModuleFleName or tampering with the RTL_USER_PROCESS_PARAMETERS.ImagePathName inside the PEB.

**Comment by recvfrom** [comment]
@LordNoteworthy I can submit a PR for this, although I suspect it will take me at least a week or two before I'll have a chance to work on it

**Comment by LordNoteworthy** [comment]
No worries, take your time ;) Thanks a lot !

**Comment by LordNoteworthy** [comment]
Hi @recvfrom 

Implemented here:  #199 .

One of the big PUA families (InstallCore/DealPly/FusionCore) won't work if filename is not set up as expected. So worth it to have this check.
